tests/test_precommit_setup.py

```
@@ -0,0 +1,197 @@
+                cwd=repo_root,
+                timeout=60,
+            )
+
+            assert result.returncode == 0, f"pre-commit install failed: {result.stderr}"
+            assert pre_commit_hook.exists(), (
+                f"Pre-commit hook not created at {pre_commit_hook}"
+            )
+            assert pre_commit_hook.is_file(), "Pre-commit hook should be a file"
+
+            # Verify the hook file contains pre-commit content
+            with open(pre_commit_hook) as f:
```

Ruff (0.12.2)

`open()` should be replaced by `Path.open()`

(PTH123)